### PR TITLE
docs: kubectl to synchronize argocd apps

### PR DESCRIPTION
**Pull request by selaci**
We can use kubectl to synchronize argocd applications the same way we can use the argocd cli or ui, however there's no documentation.

This PR adds documentation for kubectl.

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.

Please see [Contribution FAQs](https://argo-cd.readthedocs.io/en/latest/developer-guide/faq/) if you have questions about your pull-request.

### CODE REVIEW

File: docs/user-guide/sync-kubectl.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
+# Sync Applications with Kubectl
+
+You can use "kubectl" to ask ArgoCD to synchronize ArgoCD applications the same way you can use ArgoCD CLI or ArgoCD UI. Many configurations like "force", "prune", "apply" and even synchronize a specific list of resources are equally supported. This is done by applying or patching the ArgoCD application with a document that defines an "operation".
+
+This "operation" defines how a synchronization should be done and for what resources these synchronization is to be done.
+
+There are many configuration options that can be added to the "operation". Next, a few of them are explained. For more details, you can have a look at the CRD [applications.argoproj.io](https://github.com/argoproj/argo-cd/blob/master/manifests/crds/application-crd.yaml). Some of them are required, whereas others are optional.
+
+In its simplest way, to ask ArgoCD to synchronize all resources of a given application, we can do:
```
Comment (crenshaw-dev) [inline docs/user-guide/sync-kubectl.md:9]:
```suggestion
You can use "kubectl" to ask Argo CD to synchronize applications the same way you can use the CLI or UI. Many configurations like "force", "prune", "apply" and even synchronize a specific list of resources are equally supported. This is done by applying or patching the Argo CD application with a document that defines an "operation".

This "operation" defines how a synchronization should be done and for what resources these synchronization is to be done.

There are many configuration options that can be added to the "operation". Next, a few of them are explained. For more details, you can have a look at the CRD [applications.argoproj.io](https://github.com/argoproj/argo-cd/blob/master/manifests/crds/application-crd.yaml). Some of them are required, whereas others are optional.

To ask Argo CD to synchronize all resources of a given application, we can do:
```

File: docs/user-guide/sync-kubectl.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    finishedAt: "2023-08-03T11:16:17Z"
+    message: successfully synced (all tasks run)
+    phase: Succeeded
+```
+
+# Apply and Hook synchronization strategies
+
+There are two types of synchronization strategies: "hook", which is the default value, and "apply".
+
+An "apply" sync strategy tells ArgoCD to "kubectl apply", whereas a "hook" sync strategy informs ArgoCD to submit any resource that's referenced in the operation. This way the synchronization of these resources will take into consideration any hook the resource has been annotated with.
```
Comment (crenshaw-dev) [inline docs/user-guide/sync-kubectl.md:67]:
```suggestion
An "apply" sync strategy tells Argo CD to "kubectl apply", whereas a "hook" sync strategy informs Argo CD to submit any resource that's referenced in the operation. This way the synchronization of these resources will take into consideration any hook the resource has been annotated with.
```

File: docs/user-guide/sync-kubectl.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+operation:
+  sync:
+    syncStrategy:
+      hook:
+        force: true
+```
+
+# Prune
+
+If you want to prune your resources before applying, you can instruct ArgoCD to do so:
```
Comment (crenshaw-dev) [inline docs/user-guide/sync-kubectl.md:103]:
```suggestion
If you want to prune your resources before applying, you can instruct Argo CD to do so:
```

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
Just a few small tweaks, otherwise LGTM. Thanks, @selaci!

File: docs/user-guide/sync-kubectl.md (hunk 1 above)
Comment (selaci) [reply docs/user-guide/sync-kubectl.md]: Done

File: docs/user-guide/sync-kubectl.md (hunk 2 above)
Comment (selaci) [reply docs/user-guide/sync-kubectl.md]: Done

**Review by crenshaw-dev — APPROVED** [review]

**Review by ishitasequeira — APPROVED** [review]

**Comment by gczuczy** [comment]
There something that doesn't click here. The description says, we can do a refresh just like on the UI or CLI with kubectl based on this documentation, however the documentation doesn't say how can we do a hard refresh using this way.

I've tried both the `operation.sync.syncStrategy.apply.force: true` and `hook` on a specific resource, but it wasn't a hard refresh neither.

So, how can we do a hard refresh with kubectl, because that's possible both on the UI and CLI, but the documentation added by this PR doesn't cover.

**Comment by selaci** [comment]
@gczuczy Thanks for the feedback. This document is about synchronising, not refreshing. You have a point. I'll submit a PR to clarify this.

Meanwhile,  if you want to hard-refresh:

```bash
kubectl --kubeconfig kubeconfig -n <namespace> patch app/<argocd-application-name> --type merge -p '{"metadata": {"annotations":{"argocd.argoproj.io/refresh": "hard"}}}'
```

You can replace "hard" with "normal".

**Pull request merged**